articles/integrate-tailwindcss-into-flask/index.md

```diff
@@ -0,0 +1,232 @@
+
+* Python3
+* Node.js 12.13.0 or higher
+* npm or yarn
+* Flask
+* Your Favorite text editor
+* Knowledge of using TailwindCSS
+
+## Building A Simple Flask App
+
+### Installing Flask
+
+We'll begin by creating a really simple flask server, that is capable of rendering frontend templates on our behalf. To begin we open up the terminal and install flask using python's package manager, as follows:
+
+```bash
+pip install flask
+```
+
+This will install flask neatly for us on our computer, if you already have it installed you can skip this step. You can also choose to install it in a virtual environment if you feel like it.
+
+### Creating the Flask Server
+
+The next step is to create a simple flask server that will render the frontend templates for us. Create a new folder (anywhere you wish) to hold our files and create a new file called `app.py` in the folder, as well as a sub-folder called `templates`; this is where all HTML files will go, lastly we'll create another subfolder called `static`, this folder will hold all static files, including CSS files, scripts and images for our frontend templates. When you're done the folder should look like this:
+
+```bash
+projectfolder/
+   - app.py
+   - templates/
+   - static/
+```
+
+Now in app.py we'll add the following lines of code to create a server.
+
+```python
+from flask import Flask ,render_template
+
+app = Flask(__name__)
+
+@app.route("/")
+@app.route("/index")
+def index():
+	return render_template("index.html")
+
+if __name__ =='__main__':
+	app.run(debug=True)
+
+```
+
+The code block above creates a simple server with one endpoint (or route) that simple renders (or displays) a web page, which is supposed to be from a file called `index.html` as seen in the code. However, we haven't created that file yet. Remember, I mentioned that all HTML files go in the templates folder we created earlier. Hence, we'll go to that folder and add a new file called `index.html` so our flask server can see it. In `index.html` we can add a simple "hello world" h1 tag as follows:
+
+```hmtl
+<!DOCTYPE html>
+<html>
+<head>
+	<title></title>
+</head>
+<body>
+	<h1>Hello World</h1>
+</body>
+</html>
+```
+
+We'll save that and test run our server, from the terminal using the following command.
+
+```bash
+python app.py
+```
+Once the server is running head over to 
+
+`http://localhost:5000/index` to test the server and you should see the "hello world" from the `index.html` file.
+
+![demo1](/engineering-education/integrate-tailwindcss-into-flask/demo1.jpeg)
+
+## Installing TailwindCss
+Now that our server is up and running, let's install Tailwindcss. To install tailwind we can use either yarn or npm (depending on which one you have on your system). We  navigate to the `static` folder we created earlier from the terminal using the `cd` command, once there we'll install TailwindCss in there, as follows:
```

Comment:
I have a problem at this step. When I navigate to the static folder and run _npm install tailwindcss_ nothing happens. Also, don't I need to run maybe _npm init_ first so that I can install other dependencies?? (From my previous js knowledge)

Reply:
> Good morning @Curiouspaul1 I made some changes please pull them before you go on.
> Also, note that the hero image should just be named '_hero_'.
> 
> I have added the front matter, help me fill it by adding your full name in the author section.
> 
> The article is looking good now, ready for final review. Amazing work. Finish on the changes from the tech review and we will be through!

Thanks a lot boss, I'll get it done